### PR TITLE
Lazy Loading, clinit bug.

### PR DESCRIPTION
**Pull request by Degiorgio**
This pr, fixes a bug that occurs when lazy-loading flag is specified
Essentially, the availability of an instance to a class is tied together
with the need for clinit, this causes lazyloading to forgo clinit
if an instance is not created even though there are valid scenarios
were an instance is not available yet clinit is still needed.

Example:
```
public class A
{
  // verifying  this function with jbmc and the lazyloading flag specified. 
  // will result in verification failed.
  // The commit in this PR addresses this issue.
  public void me()
  {
    assert(G.x == 100);
  }
  // this always worked, since there is an instance of G  and thus lazy-loading will
  // load G.<clinit>
  public void me2()
  {
    G g = new G();
    assert(G.x == 100);
  }
}

class G
{
  static int x = 100;
}
```

### CODE REVIEW

File: src/java_bytecode/java_bytecode_convert_method.cpp [hunk 1]
```diff
@@ -1028,6 +1028,12 @@ exprt java_bytecode_convert_methodt::get_or_create_clinit_wrapper(
     code_function_callt call_real_init;
     call_real_init.function()=findsymit->second.symbol_expr();
     init_body.move_to_operands(call_real_init);
+
+    if(lazy_methods)
+      // lazy methods enabled and the clinit
```
Comment (smowton) [inline src/java_bytecode/java_bytecode_convert_method.cpp]: This comment doesn't say what's happening here. This is requesting *this* clinit be loaded; is the call to a parent clinit missing, in which case that's a separate bug?

File: src/java_bytecode/java_bytecode_convert_method.cpp [hunk 2]
```diff
@@ -1028,6 +1028,12 @@ exprt java_bytecode_convert_methodt::get_or_create_clinit_wrapper(
     code_function_callt call_real_init;
     call_real_init.function()=findsymit->second.symbol_expr();
     init_body.move_to_operands(call_real_init);
+
+    if(lazy_methods)
```
Comment (smowton) [inline src/java_bytecode/java_bytecode_convert_method.cpp:1032]: Braces around multi-line block please

**Review by smowton — COMMENTED** [review]
Comment might need fixing

File: src/java_bytecode/java_bytecode_convert_method.cpp (hunk 1 above)
Comment (Degiorgio) [reply src/java_bytecode/java_bytecode_convert_method.cpp]: yep sorry, there is is no additional bug, as   get_or_create_clinit_wrapper, will recurse on the parents of the class in question.

File: src/java_bytecode/java_bytecode_convert_method.cpp (hunk 2 above)
Comment (Degiorgio) [reply src/java_bytecode/java_bytecode_convert_method.cpp:1032]: done.

**Review by smowton — APPROVED** [review]

**Review by cesaro — APPROVED** [review]

**Comment by peterschrammel** [comment]
@Degiorgio, appveyor is failing...

**Comment by peterschrammel** [comment]
@mgudemann, please create a TG bump for this.

**Comment by Degiorgio** [comment]
@peterschrammel, will address this  later today.

**Review by thk123 — APPROVED** [review]

**Comment by NlightNFotis** [comment]
Hello.

I am closing this PR as it appears to be stale and no longer congruent with the current state of the repository (`jbmc` is under its own directory, and `java_bytecode_convert_method.cpp` does not contain the modified method at all).

If you believe this PR is adding value and would still like to see it merged, please feel free to reopen the PR and rebase it on branch `develop`.

**Pull request closed without merging**